source/hash_table.c

```diff
@@ -642,8 +650,17 @@ bool aws_hash_table_eq(
     return true;
 }
 
+/**
+ * Given an iterator, and a start slot, find the next available filled slot if it exists
+ * Otherwise, return an iter that will return true for aws_hash_iter_done().
+ * Note that aws_hash_iter_is_valid() need not hold on entry to the function, since
+ * it can be called on a partially constructed iter from aws_hash_iter_begin().
+ *
+ * Note that calling this on an iterator which is "done" is idempotent: it will return another
+ * iterator which is "done".
+ */
 static inline void s_get_next_element(struct aws_hash_iter *iter, size_t start_slot) {
-
+    AWS_PRECONDITION(iter != NULL);
```

Comment:
No need for `AWS_PRECONDITION(aws_hash_iter_is_valid(iter))`?

Reply:
See the comment in the code right above.